Setup Initial Database

raouf-kamel opened the pull request:
## 🛠️ Summary

Briefly explain what this pull request does.
- Added Room dependencies
- Added Koin dependencies
-  Created CategoryEntity and TaskEntity for Room database
- Added type converters (e.g., for handling custom data types like lists, dates, enums)
-  Implemented CategoryDao and TaskDao for accessing and querying database records
- Added mappers to convert between database entities and domain models
- Created TasksServiceImpl as the service layer implementation for task-related operations
- Configured the Room database class
- Wired everything using Koin modules

---

## 📌 Description

Please include a summary of the changes and the related issue.

 
---

## ✅️Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Code style (formatting, new lines, etc.)
- [ ] Refactoring (no functional changes, renaming)
- [x] Other (describe the changes):

## ✅ Checklist:

- [x] The PR is to the `develop` branch
- [x] My code follows the style and architecture of the project
- [ ] I have performed a self-review of my code
- [ ] I tested the application on my computer and does everything as expected

---

## 🔍 Testing

Please check what has been verified before merging:

- [x] Code compiles successfully
- [x] Unit tests (if any) pass
- [x] Feature behaves as expected
- [ ] Manually tested the affected logic

---
---

## 📎 Related Issues / Tasks

Link to any related issue or task:

- Resolves: `#issue_number` (if applicable)

---

## 👥 Reviewers

Optional: Tag team members you want to review the changes  
Example: `@teammate1` `@leadDev`

raouf-kamel committed:
add dependencies room database

raouf-kamel committed:
add dependencies koin

raouf-kamel committed:
setup initial database structure

AimanYosofiUlt reviewed (commented):
Thank you for your hard work
Though maybe you shouldn't create a pull request yet

raouf-kamel committed:
refactor: removed 'datasource' from package path to simplify structure

raouf-kamel committed:
Merge remote-tracking branch 'origin/dev' into feature/database-setup

raouf-kamel committed:
add pull request template

raouf-kamel committed:
add Category entity and task entity and converters with DAO and mapper files

raouf-kamel committed:
add task Services

mostafa-n3ma committed:
add,
- category service implementation
update, category service interface
- category dao
- rename to category local dto
- category mapper
- converters to use local date time
- add migration config to data base
- using local date time
- update the services implementation

mostafa-n3ma committed:
Revert "add,"

This reverts commit 1274cfa8e09bfc0b718d65714bb0be672ea91514.

mostafa-n3ma committed:
add,
- category service implementation
update, category service interface
- category dao
- rename to category local dto
- category mapper
- converters to use local date time
- add migration config to data base
- using local date time
- update the services implementation

raouf-kamel committed:
add bundles to room,koin

raouf-kamel committed:
rename MyApplication to TudeeAssistantApplication

raouf-kamel committed:
add ColumnInfo to all Column

raouf-kamel committed:
change createdAt column type from LocalDate to LocalDateTime

raouf-kamel committed:
Merge remote-tracking branch 'origin/feature/database-setup' into feature/database-setup

raouf-kamel committed:
add ColumnInfo to description Column

raouf-kamel committed:
rename CategoryEntity,TaskEntity to CategoryLocalDto,TaskLocalDto

raouf-kamel committed:
remove viewModelModule

raouf-kamel committed:
rename package entity to dto

mostafa-n3ma committed:
Merge branch 'feature/database-setup' into add-category-service-and-update

mostafa-n3ma committed:
Merge pull request #49 from SanaaTeam/add-category-service-and-update

add, - category service implementation

mostafa-n3ma committed:
refactor,  entity to local dto

mostafa-n3ma committed:
add category service implementation

raouf-kamel committed:
Merge pull request #51 from SanaaTeam/50-add-category-service-implementation

add category service implementation

mostafa-n3ma committed:
create, assets file and add default categories images

mostafa-n3ma committed:
add, default images resources to assets/category directory

raouf-kamel committed:
Merge remote-tracking branch 'origin/dev' into feature/database-setup

# Conflicts:
#	app/build.gradle.kts
#	app/src/main/java/com/sanaa/tudee_assistant/domain/service/CategoryService.kt
#	app/src/main/java/com/sanaa/tudee_assistant/domain/service/TaskService.kt

raouf-kamel committed:
rename TasksServices to TaskService

mostafa-n3ma committed:
add, assets image loader function load images from assets file
create default images list for call back use

mostafa-n3ma committed:
- add, default categories list from assets
- update, database di module with koin
- add default categories call back to database
- add, image loader from assets path

ZainabAlnajjar reviewed (changes requested):
good job, just check on my comments

mostafa-n3ma committed:
delete unnecessary imports and clean up

mostafa-n3ma committed:
update , database set up

mostafa-n3ma committed:
update ,imagePath for domaine model

mostafa-n3ma committed:
refactor

mostafa-n3ma committed:
Merge pull request #59 from SanaaTeam/56-add-ssets-file-implementation-and-add-default-data-via-room-call-back

56-add-assets-file-implementation-and-add-default-data-via-room-call-back

mostafa-n3ma committed:
Improve database error handling in Task and Category services

- Add consistent exception handling for all DB operations
- Convert all Flow queries to use proper error mapping
- Standardize CRUD operation failure responses

mostafa-n3ma committed:
add , default category exception

mostafa-n3ma committed:
fix, bugs in mapper and domain classes so as the dto
update in delete all categories

mostafa-n3ma committed:
Merge pull request #65 from SanaaTeam/64-implement-proper-exception-handling-in-tasks-and-categories-services

Enhanced Database Error Handling

mostafa-n3ma committed:
Merge remote-tracking branch 'origin/dev' into feature/database-setup

# Conflicts:
#	app/src/main/java/com/sanaa/tudee_assistant/domain/service/CategoryService.kt
#	app/src/main/java/com/sanaa/tudee_assistant/domain/service/TaskService.kt

mostafa-n3ma committed:
resolve conflict

AimanYosofiUlt reviewed (approved):
Good Job, Thank you for your work and the offer to resolve the issues
